### PR TITLE
Better dev environment, TS clean, new capabilities

### LINKED ISSUE #27: Use npm-run-all instead of gulp

### LINKED ISSUE #40: Question the `prepublish`script
Since there is a `prepublish` script that compiles basically everything, dependencies are not needed...
Is it a good choice ?

### LINKED ISSUE #42: The button retry is far too expensive.
Yesterday we had a bug on 6element and bin couldn't be retrieved. So we changed stuf and pressed many times on the retry button. But this button can be pressed so easily that it costed us 9.5euros:

screen shot 2016-01-06 at 09 14 13

I think we should at least impose a limit to number of times it can be pressed per minutes.

### PR DESCRIPTION
**Pull request by bleucitron**
This PR includes:

**New stuff**
- The icons are now bigger
- The RetryButton triggers a 10s timeout before being functional again to avoid spamming. Fixes #42 
- `BinServer` is now an ES6 class, `path` and `port` are passed as parameters.
- Route `/waste-categories` is opened to allow access to `waste-categories` peer dependency from `6bin` server's filesystem. There still might be a better way=> @DavidBruant check #39  

**Typescript**
- All TS errors are now solved.
- The definition file `tsd.d.ts` is versioned.
- Useless definitions files are removed

**Environment**
- Test command line fixed
- `gulp` replaced by `npm-run-all`. Fixes #27 
- We don't need to update 6bin npm version when waste-categories' version is updated. Fixes #40

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,12 +58,10 @@
     "socket.io-client": "^1.3.7",
     "svg-injector": "^1.1.3",
     "vinyl-source-stream": "^1.1.0",
-    "waste-categories": "^1.0.4"
+    "waste-categories": "^1.0.4",
+    "watch-exec": "^1.2.0"
```
Comment (DavidBruant) [inline package.json]: Is it used?

File: ts/client/Components/Dumb/SVG.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 import * as React from 'react';
 import { findDOMNode, unmountComponentAtNode } from 'react-dom';
-import * as SVGInjector from 'svg-injector';
+import SVGInjector = require('svg-injector');
```
Comment (DavidBruant) [inline ts/client/Components/Dumb/SVG.ts:5]: ?

File: package.json (hunk 1 above)
Comment (bleucitron) [inline package.json]: Nope

File: ts/client/Components/Dumb/SVG.ts (hunk 2 above)
Comment (bleucitron) [inline ts/client/Components/Dumb/SVG.ts:5]: This is TS related. I struggled to make it work without TS errors. This is the pattern i found, if you have better options, let me know about it.

**Pull request merged**